components/settings/account/components/NewIdentityModal.tsx

```
@@ -91,6 +93,16 @@ export function NewIdentityModal({ isOpen, onClose }: Props) {
       sendingMagicLink.current = true;
       // console.log('Handling magic link request');
       try {
+        if (space) {
+          const isUserBannedFromSpace = await charmClient.members.checkSpaceBanStatus({
+            spaceId: space.id,
+            email
+          });
+
+          if (isUserBannedFromSpace.isBanned) {
```

it's nice to be proactive, but in the name of simplicity i would have just put this logic inside loginWithMagicLink() and let it fail when they try to use the link

---

Yep we are now taking a more reactive approach. Its definitely a lot simpler now.